features/flows/classification/src/main/resources/OSGI-INF/blueprint/blueprint.xml

```diff
@@ -0,0 +1,31 @@
+<blueprint xmlns="http://www.osgi.org/xmlns/blueprint/v1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:cm="http://aries.apache.org/blueprint/xmlns/blueprint-cm/v1.1.0"
+           xmlns:ext="http://aries.apache.org/blueprint/xmlns/blueprint-ext/v1.1.0"
+           xsi:schemaLocation="
+		http://www.osgi.org/xmlns/blueprint/v1.0.0
+		http://www.osgi.org/xmlns/blueprint/v1.0.0/blueprint.xsd
+
+		http://aries.apache.org/blueprint/xmlns/blueprint-cm/v1.1.0
+		http://aries.apache.org/schemas/blueprint-cm/blueprint-cm-1.1.0.xsd
+
+		http://aries.apache.org/blueprint/xmlns/blueprint-ext/v1.1.0
+		http://aries.apache.org/schemas/blueprint-ext/blueprint-ext-1.1.xsd
+">
+
+    <!-- DAO -->
+    <bean id="classificationRuleDao" class="org.opennms.netmgt.flows.classification.persistence.DummyClassificationRuleDAO"/>
+
+    <!-- Classification Engine -->
+    <bean id="defaultClassificationEngine" class="org.opennms.netmgt.flows.classification.DefaultClassificationEngine">
+        <argument ref="classificationRuleDao" />
+    </bean>
+    <bean id="cachingClassificationEngine" class="org.opennms.netmgt.flows.classification.CachingClassificationEngine">
+        <argument ref="defaultClassificationEngine" />
+    </bean>
+    <bean id="threadSafeClassificationEngine" class="org.opennms.netmgt.flows.classification.ThreadSafeClassificationEngine">
+        <argument ref="cachingClassificationEngine" />
```

Comment:
The cache is already thread safe, shouldn't we wrap the DefaultClassificationEngine instead?

Reply:
Yes, true.